tests/OnTypeFormatting.test.cpp

```diff
@@ -0,0 +1,119 @@
+#include "doctest.h"
+#include "Fixture.h"
+#include "LSP/LanguageServer.hpp"
+
+TEST_SUITE_BEGIN("OnTypeFormatting");
+
+TEST_CASE_FIXTURE(Fixture, "on_type_formatting_converts_quotes")
+{
+    auto uri = newDocument("file.luau", "print(\"aaa {\")");
+
+    lsp::DocumentOnTypeFormattingParams params;
+    params.textDocument.uri = uri;
+    params.position = {0, 12}; // after the '{'
+    params.ch = "{";
+    params.options = {4, true};
+
+    auto edits = workspace.onTypeFormatting(params);
+    REQUIRE(edits.has_value());
+    REQUIRE(edits->size() == 2);
+
+    CHECK_EQ(edits->at(0).newText, "`");
+    CHECK_EQ(edits->at(1).newText, "`");
+}
+
+TEST_CASE_FIXTURE(Fixture, "on_type_formatting_converts_quotes_in_unfinished_string")
+{
+    auto uri = newDocument("file.luau", "print(\"aaa {");
+
+    lsp::DocumentOnTypeFormattingParams params;
+    params.textDocument.uri = uri;
+    params.position = {0, 12}; // after the '{'
+    params.ch = "{";
+    params.options = {4, true};
+
+    auto edits = workspace.onTypeFormatting(params);
+    REQUIRE(edits.has_value());
+    REQUIRE(edits->size() >= 1);
```

Comment:
```suggestion
    REQUIRE(edits->size() == 2);
```

Or are we expecting more than 2 edits? Same in the code down below

Also, not sure if it helps to visualise better, but could extract the `applyEdit` function out of Rename.test.cpp to test the final result.

Reply:
> but could extract the applyEdit function out of Rename.test.cpp to test the final result.

Yes it helps to visualize better the result